openslides_backend/action/relations/relation_manager.py

```diff
@@ -43,6 +43,40 @@ def get_relation_updates(
         self.process_template_fields(model, instance)
 
         relations: RelationUpdates = {}
+        for field_name in instance:
+            if not model.has_field(field_name):
+                continue
+            field = model.get_field(field_name)
+
+            # only relations are handled here
+            if not isinstance(field, BaseRelationField):
+                continue
+            # ignore template fields, we have to do no relation handling there
+            if isinstance(field, BaseTemplateField) and field.is_template_field(
+                field_name
+            ):
+                continue
+
+            handler = SingleRelationHandler(
+                self.datastore,
+                field,
+                field_name,
+                instance,
+            )
+            result = handler.perform()
```

Comment:
please deduplicate, this doubles with the function below

Reply:
also, calling the `SingleRelationHandler` twice for each field seems very bad, performance-wise, especially since we know that the relation handling takes ost of the action time. See if you can restructure it such that it is only called once

Reply:
Restructured this code and use the SingleRelationHandler only once.